cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/template/etl/batch/ETLMapReduceTest.java

```diff
@@ -170,4 +179,56 @@ public void testTableToTable() throws Exception {
     Assert.assertEquals("island", row.getString("item"));
   }
 
+  @Test
+  public void testS3toTPFS() throws Exception {
+    String testPath = "s3n://test/2015-06-17-00-00-00.txt";
```

Comment:
I'm not sure how s3 works, but is this a valid URI? the component between the second and third slashes is supposed to be an authority (host:port). I guess its ok coz you're using an in-memory s3?